src/olympia/versions/tasks.py

```diff
@@ -112,11 +113,95 @@ def delete_preview_files(pk, **kw):
 
 @task
 @use_primary_db
-def extract_version_to_git(version_id, author_id=None, note=None):
-    """Extract a `File` into our git storage backend."""
+def extract_addon_to_git(addon_pk):
+    addon = Addon.unfiltered.get(pk=addon_pk)
+    log.info(
+        'Starting extraction of addon "{}" to git storage.'.format(addon_pk)
+    )
+
+    if addon.git_extraction_is_in_progress:
+        log.info('Aborting extraction of addon "{}" to git storage because it '
+                 'is already in progress.'.format(addon_pk))
+        return
+
+    # "Lock" this add-on so that the git extraction can only run once and tasks
+    # for extracting related versions are delayed until this task has finished.
+    git_storage, created = GitExtraction.objects.update_or_create(
+        addon=addon,
+        defaults={'in_progress': True},
+    )
+
+    # Filter out versions that are already present in the git storage.
+    versions = addon.versions.filter(git_hash='').order_by('created')
+
+    for version in versions:
+        try:
+            log.info('Starting extraction of version "{}" for addon "{}" to '
+                     'git storage.'.format(version.pk, addon_pk))
+
+            extract_version_to_git(
```

Comment:
This should be called with `.delay()`, otherwise you're directly calling the function. You'll end up hitting time-limits if not using tasks... See comment about `chain` below.

Reply:
Again, this was borrowed from the migrate command. Calling the function directly was not a mistake here but I forgot about the soft timeout...

Reply:
Interesting. I am curious to see if we did hit timeouts when we ran the migration... I don't think it's safe to bet that we'll never hit it, but maybe...